execute/tokendata/usdc/attestation.go

```diff
@@ -32,8 +35,8 @@ type AttestationStatus struct {
 type AttestationClient interface {
 	Attestations(
 		ctx context.Context,
-		msgs map[cciptypes.ChainSelector]map[cciptypes.SeqNum]map[int][]byte,
-	) (map[cciptypes.ChainSelector]map[cciptypes.SeqNum]map[int]AttestationStatus, error)
+		msgs map[cciptypes.ChainSelector]map[exectypes.MessageTokenID]reader.MessageHash,
+	) (map[cciptypes.ChainSelector]map[exectypes.MessageTokenID]AttestationStatus, error)
```

Comment:
Moving away from `map[int][]byte` means you'll need to update the `ValidateObservation` function to check that you don't have `MessageTokenID`'s.

Reply:
This type is used only as an intermediate step in the usdc processing. There are no changes to how observations look like, no changes to the business logic required as well. Am I missing sth?

Reply:
I must be the one missing something. Just want to make sure we don't have illegal observations.

We have a problem in some types like this:
```
some_observation_data: []{
   { chain:1, data:...},
   { chain:2, data:...},
   { chain:1, data:...}, // illegal but not enforced by the type system so we have to validate manually
}
```

Reply:
I don't think we introduced any regression here, but additional tests and wiring up together with ocr is going to be another pr ;)